app/src/main/java/uk/ac/ebi/atlas/experimentpage/tsneplot/TSnePlotSettingsService.java

```diff
@@ -32,10 +32,15 @@ public TSnePlotSettingsService(DataFileHub dataFileHub,
     }
 
     public List<Integer> getAvailableKs(String experimentAccession) {
-        try (TsvStreamer clustersTsvStreamer =
-                     dataFileHub.getSingleCellExperimentFiles(experimentAccession).clustersTsv.get()) {
+        var clustersTsv = dataFileHub.getSingleCellExperimentFiles(experimentAccession).clustersTsv;
```

Comment:
You can DRY this code by extracting it to a well named private method as you are using it at least another place, too.
Thanks